src/utils/player-error.js

```diff
@@ -1,38 +1,802 @@
 //@flow
-export default class PlayerError {
-  static TYPE: {[name: string]: Object} = {
-    NOT_REGISTERED_PLUGIN: {
-      name: "PluginNotRegisteredException",
-      message: function (name) {
-        return `Cannot load ${name} plugin. Name not found in the registry`;
-      }
-    },
-    NOT_VALID_HANDLER: {
-      name: "PluginHandlerIsNotValidException",
-      message: function () {
-        return "To activate plugin you must provide a class derived from BasePlugin";
+
+/**
+ * @classdesc This is a description of the MyClass class.
+ */
+export default class PlayerError{
+
+  /**
+   * @constructor
+   */
+  constructor(){
+  }
+
+  /**
+   * builds an error string from the error object
+   * @param {any} error - error object
+   * @returns {string} - a readable error message
+   */
+  static createError(error: any): string{
+    const severity = error.severity;
+    const category = error.category;
+    const code = error.code;
+    const data = JSON.stringify(error.args);
+    let severityName = severity === 1 ? 'RECOVERABLE' : 'CRITICAL';
```

Comment:
I think we should add the explicit data only when in logLevel DEBUG.
This way we can only run this when in debug mode and expose the strings only then, otherwise just use the codes.